### PR TITLE
fix(config): load all default options not in existing config

### PR DESCRIPTION
**Pull request by Audionut**
will self populate new options

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved configuration loading by overlaying embedded defaults onto stored settings before decrypting secrets, while preserving explicitly stored `false`/`0`/empty values.
  * More robust tracker default merging (including case-insensitive handling) with correct, persisted repairs and safer handling of invalid or ambiguous tracker data.
  * Prevents partial write-back on repair failures and now rejects malformed saved JSON that contains duplicate keys.

* **Tests**
  * Expanded coverage for default backfill/repair reporting, rollback behavior, tracker merge edge cases, and duplicate-key rejection.

### CODE REVIEW

**Comment by Audionut** [comment]
@coderabbitai review

File: internal/configstore/dbloader_test.go [hunk 1]
```diff
@@ -92,6 +92,68 @@ func TestLoadFromDBPathBackfillsMissingTrackerDefaults(t *testing.T) {
...
+
+	loaded, err := configstore.LoadFromDBPath(ctx, dbPath)
+	if err != nil {
+		t.Fatalf("load config from database: %v", err)
+	}
+	if got := loaded.PostUpload.MaxConcurrentTrackers; got != 4 {
+		t.Fatalf("expected max concurrent tracker uploads default, got %d", got)
+	}
+	if loaded.PostUpload.CrossSeeding {
+		t.Fatal("expected explicit cross_seeding=false to be preserved")
```
[the remark this answers is not in the document]
Comment (Audionut) [reply internal/configstore/dbloader_test.go:287]: configstore.LoadFromDBPath does apply env overrides, but internal/config/env.go:14 does not override PostUpload.MaxConcurrentTrackers or PostUpload.CrossSeeding. Current test assertions at internal/configstore/dbloader_test.go:117 are not env-sensitive for those fields.

  Validation:

  - go test ./internal/configstore -run TestLoadFromDBPathBackfillsMissingStoredOptions -count=1

**Comment by Audionut** [comment]
@coderabbitai review

**Comment by Audionut** [comment]
@coderabbitai review
